File: clients/ios/PandaSolve/Data/UserRepository.swift
import Foundation

final class UserRepository {
    private let api: APIClient
    /// Last loaded profile, kept so screens can show counts instantly on re-open.
    private(set) var lastMe: Me?

    init(api: APIClient) { self.api = api }

    func me() async throws -> Me {
        let m: Me = try await api.get("/v1/me")
        lastMe = m
        return m
    }

    func startLink() async throws -> String {
        let resp: LinkStartResponse = try await api.post("/v1/auth/link/start", body: EmptyBody())
        return resp.code
    }

    func updateLanguage(_ code: String) async throws -> Me {
        try await api.post("/v1/me", body: UpdateMeRequest(languageCode: code))
    }

    func updateDisplayName(_ name: String) async throws -> Me {
        try await api.post("/v1/me", body: UpdateMeRequest(displayName: name))
    }
}

struct EmptyBody: Codable {}

File: clients/android/app/src/main/java/com/pandasolve/app/data/repository/UserRepository.kt
package com.pandasolve.app.data.repository

import com.pandasolve.app.domain.model.Me
import com.pandasolve.app.domain.model.UpdateMeRequest
import com.pandasolve.app.network.PandaApiService
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class UserRepository @Inject constructor(
    private val api: PandaApiService,
) {
    /** Last successfully-loaded profile, kept in memory so screens can show counts
     *  instantly on re-open instead of waiting for a fresh /v1/me. */
    @Volatile
    var lastMe: Me? = null
        private set

    suspend fun me(): Me = api.getMe().also { lastMe = it }

    suspend fun startLink(): String = api.startLink().code

    suspend fun updateLanguage(code: String): Me =
        api.updateMe(UpdateMeRequest(languageCode = code))

    suspend fun updateDisplayName(name: String): Me =
        api.updateMe(UpdateMeRequest(displayName = name))
}
